holy-grail-frontend/src/features/Developer/DeveloperAddSubjectModal.tsx

```diff
@@ -0,0 +1,89 @@
+import { useForm } from 'react-hook-form';
+import { DeveloperAddModalProps, singularDataType, AddSubjectDetails } from '@features';
+import { AlertToast, AlertProps, Combobox } from '@components';
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  TextField,
+  Stack,
+} from '@mui/material';
+import { createSubject } from '@api/actions';
+import { useEffect, useState } from 'react';
+import { fetchData, CategoryType } from '@api/library';
+
+export const DeveloperAddSubjectModal = ({
+  isOpen,
+  onClose,
+  onSuccessfulAdd,
+}: Omit<DeveloperAddModalProps, 'type'>) => {
+  const singularType = singularDataType['subjects'];
+  const { register, handleSubmit } = useForm<AddSubjectDetails>();
+
+  const [openAlert, setOpenAlert] = useState<boolean>(false);
+  const [alertContent, setAlertContent] = useState<AlertProps | undefined>(undefined);
+
+  const [categoryData, setCategoryData] = useState<CategoryType[]>([]);
+
+  useEffect(() => {
+    fetchData().then(({ categories }) => {
+      setCategoryData(categories);
+    });
+  }, []);
+
+  const handleAdd = async (formData: AddSubjectDetails) => {
+    try {
+      await createSubject(formData);
+
+      await onSuccessfulAdd();
+    } catch (err) {
+      console.error(err);
+    } finally {
+      onClose();
+    }
+  };
+
+  return (
+    <>
+      <Dialog open={isOpen} onClose={onClose}>
+        <DialogTitle>Add {singularType}</DialogTitle>
+        <form onSubmit={handleSubmit(handleAdd)}>
+          <DialogContent>
+            <DialogContentText marginBottom='10%'>Create new {singularType}.</DialogContentText>
+            <Stack direction='column' spacing={2}>
+              <TextField
+                {...register('name', { required: true })}
+                autoFocus
+                margin='dense'
+                label={<span style={{ textTransform: 'capitalize' }}>{singularType}</span>}
+                type='text'
+                fullWidth
+              />
+              <Combobox
+                label='Select a category'
+                options={categoryData}
+                onChange={() => {}}
```

Comment:
not sure how to use ``register`` here, given that combobox types ``onchange`` as ``(newValue: number | '') => void`` and not a ``React.ChangeEvent``. Could use mui's raw ``Autocomplete`` instead, but i'm not sure how to handle ``Commontype`` as an argument here.